rdmo/projects/progress.py

```diff
@@ -158,12 +169,22 @@ def count_questions(element, sets, conditions):
                 # for optional questions add just the number of present answers, so that
                 # only answered questions count for the progress/navigation
                 # use the max function, since the same attribute could appear twice in the tree
-                if child.attribute is not None:
+                if child.attribute:
                     if child.is_optional:
                         child_count = sum(len(set_indexes) for set_indexes in sets[child.attribute.id].values())
                         counts[child.attribute.id] = max(counts[child.attribute.id], child_count)
                     else:
-                        counts[child.attribute.id] = max(counts[child.attribute.id], set_count)
+                        resolved_condition_sets = set()
+                        condition_intersection = list(child_conditions.intersection(conditions))
+                        # update the set_count for the current child element
+                        # check for the sets that have conditions resolved to true
+                        for child_condition in condition_intersection:
+                            resolved_condition_sets.update(conditions[child_condition])
+                        if condition_intersection:
+                            current_set_count = len(counted_sets & resolved_condition_sets)
```

Comment:
this is the part that fixes the set count for a certain question, it only needs these set indexes in the conditions from before where `conditions.resolve(values)` was called